crates/swc_ecma_utils/src/function/fn_env_hoister.rs

```diff
@@ -126,57 +145,378 @@ impl FnEnvState {
+                        }
+                    }
+                };
+                e.visit_mut_children_with(self)
+            }
+            // super.foo ++
+            Expr::Update(UpdateExpr {
+                span,
+                op,
+                prefix,
+                arg,
+            }) => {
+                if let Expr::SuperProp(SuperPropExpr {
+                    span: arg_span,
+                    prop,
+                    ..
+                }) = &mut **arg
+                {
+                    let tmp = private_ident!("tmp");
+                    self.extra_ident.push(tmp.clone());
+
+                    let op = match op {
+                        op!("++") => op!(bin, "+"),
+                        op!("--") => op!(bin, "-"),
+                    };
+
+                    let update_expr = Expr::Bin(BinExpr {
+                        span: DUMMY_SP,
+                        left: Box::new(Expr::Ident(tmp.clone())),
+                        right: Box::new(Expr::Lit(Lit::Num(Number {
+                            span: DUMMY_SP,
+                            value: 1_f64,
+                        }))),
+                        op,
+                    })
+                    .as_arg();
+
+                    let mut exprs = match prop {
+                        SuperProp::Ident(id) => vec![
+                            Box::new(Expr::Assign(AssignExpr {
+                                left: PatOrExpr::Pat(Box::new(tmp.clone().into())),
+                                op: op!("="),
+                                right: Box::new(
+                                    self.super_get(&mut id.clone(), *arg_span)
+                                        .as_call(*arg_span, Vec::new()),
+                                ),
+                                span: *arg_span,
+                            })),
+                            Box::new(
+                                self.super_set(&mut id.clone(), *arg_span)
+                                    .as_call(*arg_span, vec![update_expr]),
+                            ),
+                        ],
+                        SuperProp::Computed(c) => {
+                            let prop = private_ident!("prop");
+                            self.extra_ident.push(prop.clone());
+                            vec![
+                                Box::new(Expr::Assign(AssignExpr {
+                                    span: *arg_span,
+                                    left: PatOrExpr::Pat(Box::new(tmp.clone().into())),
+                                    op: op!("="),
+                                    right: Box::new(self.super_get_computed(*arg_span).as_call(
+                                        *arg_span,
+                                        vec![Expr::Assign(AssignExpr {
+                                            span: c.span,
+                                            left: PatOrExpr::Pat(Box::new(Pat::Ident(
```

Comment:
Please use `prop.clone().into()` instead of `Box::new(Pat::Ident(prop.clone().into()))`.